### PR TITLE
chore(release): merge develop to main

### PR DESCRIPTION
**Pull request by mkolasinski-splunk**
### Description

This release include:
- upgrade tests fixes
- spl2 content tests adjustments
- conditions limiting tests execution to latest splunk
- fix for setup job to properly install dependencies in edge cases
- fix for pre-publish job to pass arguments to publish job instead of failing a workflow

### Checklist

- [ ] `README.md` has been updated or is not required
- [x] push trigger tests
- [ ] manual release test
- [ ] automated releases test
- [x] pull request trigger tests
- [x] schedule trigger tests
- [ ] workflow errors/warnings reviewed and addressed

### Testing done 
https://github.com/splunk/splunk-add-on-for-amazon-web-services/actions/runs/13651290635
https://github.com/splunk/splunk-add-on-for-amazon-web-services/actions/runs/13568681041/attempts/1
PR to main scenario: https://github.com/splunk/splunk-add-on-for-salesforce/actions/runs/13568730203
PR to develop scenario: https://github.com/splunk/splunk-add-on-for-microsoft-cloud-services/actions/runs/13568715908
release scenario: https://github.com/splunk/test-addonfactory-repo/actions/runs/13566246210

### CODE REVIEW

File: .github/workflows/reusable-build-test-release.yml [hunk 1]
```diff
@@ -48,11 +48,15 @@ on:
...
+        description: "List with TA versions (in 'X.X.X' format) that should be used as starting points for upgrade tests. Example: ['7.6.0', '7.7.0']"
         type: string
         default: >-
-          ["latest"]
+          [""]
+      wfe-run-on-splunk-latest:
+        required: false
+        description: "Forces WFE tests to run only on the latest Splunk when set to true. When set to false - will run on all supported Splunk versions required for the release. When not set - default behavior."
+        type: string
+        default: ""
```
Comment (kdoroszko-splunk) [inline .github/workflows/reusable-build-test-release.yml:59]:
```suggestion
        default: 'false'
```

**Comment by kdoroszko-splunk** [comment]
It might not be connected to this release, but should we consider setting version of `image: ghcr.io/splunk/spl2-testing-base:latest` I can recall we were avoiding using the latest tag for images used in the workflow: https://github.com/splunk/addonfactory-workflow-addon-release/blob/develop/.github/workflows/reusable-build-test-release.yml#L1234C29-L1234C46

**Review by pawel-dabro — DISMISSED** [review]

**Review by kdoroszko-splunk — APPROVED** [review]
